ddl/partition.go

```diff
@@ -334,9 +335,11 @@ func buildTablePartitionInfo(ctx sessionctx.Context, s *ast.PartitionOptions, tb
 		for _, cn := range s.ColumnNames {
 			pi.Columns = append(pi.Columns, cn.Name)
 		}
+		if err := checkColumnsPartitionType(tbInfo); err != nil {
```

Comment:
if we `checkColumnsPartitionType` here, we can remove the same check in function `checkPartitionByRange` and `checkPartitionByList`

Reply:
great, done.